src/lightning/data/streaming/data_processor.py

```diff
@@ -116,16 +110,19 @@ def _download_data_target(
         index, paths = r
 
         # 5. Check whether all the files are already downloaded
-        if all(os.path.exists(p.replace(input_dir, cache_dir) if input_dir else p) for p in paths):
+        if all(os.path.exists(p.replace(input_dir.path, cache_dir) if input_dir else p) for p in paths):
```

Comment:
They might have been downloaded but not completely, is there anything we can do to alleviate?

Reply:
We clean up the cache folder every time you start the script.